File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = Counter(s)
        for c in t:
            if c in count:
                count[c] -= 1
            else:
                return False
        
        for _, v in count.items():
            if v > 0:
                return False
        return True

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (auto i : nums) {
            if (seen.count(i) > 0) {
                return true;
            }
            seen.insert(i);
        }
        return false;
    }
};